### PR TITLE
User list and temporary delete functionality

### PR DESCRIPTION
**Pull request by Paalar**
- Adds functionality to view all students
- Adds functionality to delete all checked students, with a check all button
- Not connected to backend
- Works on #159 and #160

### CODE REVIEW

File: frontend/src/components/ViewStudents/index.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import { IStore } from '../../store';
+import Presentational from './Presentational';
+import './viewStudents.css';
+
+export interface IStateProps {
+  students?: any;
+}
+
+interface IDispatchProps {
+  someting?: any;
```
Comment (aslakhol) [inline frontend/src/components/ViewStudents/index.tsx]: Better variable name please

File: frontend/src/components/ViewStudents/index.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  // fetchStudents: () => void;
```
Comment (aslakhol) [inline frontend/src/components/ViewStudents/index.tsx]: remove comments

File: frontend/src/components/ViewStudents/index.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+type Props = IStateProps & IDispatchProps & IRouterProps;
+
+// tslint:disable-next-line:class-name
+class _Container extends Component<Props, IState> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      students: [
+        { id: 1, username: 'usrnam' },
```
Comment (aslakhol) [inline frontend/src/components/ViewStudents/index.tsx]: This is test data, please remove.

File: frontend/src/components/ViewStudents/index.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        { id: 2, username: 'putnam' },
+        { id: 3, username: 'sumnam' },
+      ],
+      usersToBeDeleted: [],
+    };
+  }
+  // public componentDidMount = () => this.props.fetchStudents();
```
Comment (aslakhol) [inline frontend/src/components/ViewStudents/index.tsx]: remove comments

File: frontend/src/components/ViewStudents/index.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+  public render() {
+    const { history } = this.props;
```
Comment (aslakhol) [inline frontend/src/components/ViewStudents/index.tsx]: This variable is not used.

File: frontend/src/components/ViewStudents/index.tsx [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    if (!usersToBeDeleted.includes(id)) {
+      this.setState(state => ({ usersToBeDeleted: [...state.usersToBeDeleted, id] }));
+    } else {
+      const filteredUsers = usersToBeDeleted.filter(userId => id !== userId);
+      this.setState({ usersToBeDeleted: filteredUsers });
+    }
+  }
+}
+
+const mapStateToProps = (state: IStore) => ({
```
Comment (aslakhol) [inline frontend/src/components/ViewStudents/index.tsx]: if mstp and mdtp are not used please remove them and use null in the connect.

File: frontend/src/components/ViewStudents/index.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+});
+
+const mapDispatchToProps = (dispatch: ThunkDispatch<{}, {}, any>) => ({
```
Comment (aslakhol) [inline frontend/src/components/ViewStudents/index.tsx]: See above

File: frontend/src/components/ViewStudents/index.tsx [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+  // fetchRooms: () => dispatch(fetchAllRooms()),
```
Comment (aslakhol) [inline frontend/src/components/ViewStudents/index.tsx]: remove comments

File: frontend/src/components/ViewStudents/index.tsx [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+});
+
+const ViewStudents = connect(
+  mapStateToProps,
```
Comment (aslakhol) [inline frontend/src/components/ViewStudents/index.tsx]: see above

File: frontend/src/components/ViewStudents/UserLink.tsx [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+        className="user-delete-checkbox"
+        label={_CHECK_CHECKBOX}
+      />
+      <LenkepanelBase
+        key={user.id}
+        linkCreator={link}
+        href={ROUTE_TO}
+      >
+        <div className="user-link">
+          <div className="link-name"><Element>{_USER_FULL_NAME}</Element>{'full name'}</div>
```
Comment (FluidSense) [inline frontend/src/components/ViewStudents/UserLink.tsx]: 'full name' should probably be user.fullname

File: frontend/src/components/ViewStudents/constants.ts [hunk 11]
```diff
@@ -0,0 +1,3 @@
+export const SUCCESFULLY_FETCHED_STUDENTS = 'SUCCESFULLY_FETCHED_STUDENTS';
```
Comment (FluidSense) [inline frontend/src/components/ViewStudents/constants.ts]: Missing an S in success

**Pull request merged**